guides/hyperlane-use-tia-for-gas.md

```diff
@@ -0,0 +1,446 @@
+# How to use Hyperlane token (TIA) as gas token in your rollup
+
+This guide builds upon [CosmWasm rollup](../tutorials/cosmwasm.md), where you set up a local rollup with integrated CosmWasm. In this guide, we will be exploring how to integrate Hyperlane as a safer IBC alterntive for token transfers in your rollup.
+
+<!-- markdownlint-disable MD033 -->
+<script setup>
+import Callout from '../.vitepress/components/callout.vue'
+</script>
+
+:::tip
+<Callout />
+:::
+
+<!-- markdownlint-enable MD033 -->
+
+## 📝 In short, deploy a CosmWasm rollup
+
+This is a gist of the [CosmWasm rollup](../tutorials/cosmwasm.md) guide.
+
+```bash
+git clone --branch v0.50.0 --depth 1 https://github.com/CosmWasm/wasmd.git
+cd wasmd
+go mod edit -replace github.com/cosmos/cosmos-sdk=github.com/rollkit/cosmos-sdk@v0.50.6-rollkit-v0.13.3-no-fraud-proofs
+go mod tidy -compat=1.17
+go mod download
+```
+
+Comment out lines 898-900 in `app/app.go`:
+
+```bash
+echo 'diff --git a/app/app.go b/app/app.go
+index 5c67ba31..797b6bf4 100644
+--- a/app/app.go
++++ b/app/app.go
+@@ -895,9 +895,6 @@ func NewWasmApp(
+ 	// At startup, after all modules have been registered, check that all proto
+ 	// annotations are correct.
+ 	protoFiles, err := proto.MergedRegistry()
+-	if err != nil {
+-		panic(err)
+-	}
+ 	err = msgservice.ValidateProtoAnnotations(protoFiles)
+ 	if err != nil {
+ 		// Once we switch to using protoreflect-based antehandlers, we might' | git apply
+```
+
+Build `wasmd`:
+
+```bash
+make install
+```
+
+To start a local DA, run:
+
+```bash
+curl -sSL https://rollkit.dev/install-local-da.sh | bash -s v0.2.0
+```
+
+From inside the `wasmd` directory, run the init script:
+
+```bash
+curl -sSL https://rollkit.dev/cosmwasm/init.sh | perl -pe 's/127\.0\.0\.1/0.0.0.0/g' | sh
+```
+
+:::info
+We modify the listening address from `127.0.0.1` to `0.0.0.0` so that it is accessible from the Hyperlane docker agents later on.
+:::
+
+With that, we have kickstarted our `wasmd` network!
+
+## 💻 Deploy the Hyperlane contracts {#deploy-contracts}
+
+Fork the cw-hyperlane repo:
+
+```bash
+git clone --depth 1 git@github.com:many-things/cw-hyperlane.git
+cd cw-hyperlane
+```
+
+#### Create `config.yaml` with our networks setup:
+
+```bash
+echo 'networks:
+  - id: "localwasmd"
+    hrp: "wasm"
+    endpoint:
+      rpc: "http://localhost:36657"
+      rest: "http://localhost:1317"
+      grpc: "http://172.17.0.1:9290"
```

Comment:
is this not supposed to be localhost / is it the docker bridge network or something?

Reply:
It allows the docker guest to access the rollup on the host. This and an additional config line on the docker compose file